### ISSUE TITLE
Another unexpected parser error

### ISSUE THREAD

**Issue by garthk** [rrrene/credo#732]
### Environment

* Credo version (`mix credo -v`): 1.1.3
* Erlang/Elixir version (`elixir -v`): Elixir 1.9.4 (compiled with Erlang/OTP 22)
* Operating system: multiple

### What were you trying to do?

`mix credo --strict` with this file:

```elixir
defmodule InflictParserError do
  @moduledoc false

  def a_method do
    ~S(")
    ~S(])
    ~S([)
  end

  @doc """
  """
  def another_method, do: nil
end
```

Our original file was much larger. I ran it through `iconv -f utf-8 -t ascii` to get rid of any #729 pirate flags, then cut it down to size.

### Expected outcome

No `Credo.Code.ParserError`

### Actual outcome

```
Unexpected error while parsing thefile.ex: [%Credo.Issue{category: :error, check: Credo.Code.ParserError, column: nil, exit_status: 0, filename: "nofilename", line_no: 14, message: "missing terminator: end (for \"do\" starting at line 1)", meta: [], priority: 0, scope: nil, severity: nil, trigger: nil}]
  lib/credo/check/readability/string_sigils.ex:50: Credo.Check.Readability.StringSigils.run/2
  lib/credo/check/runner.ex:168: Credo.Check.Runner.do_run_check_on_single_source_file/3
  lib/credo/check/runner.ex:161: Credo.Check.Runner.run_check_on_single_source_file/3
  (elixir) lib/enum.ex:1336: Enum."-map/2-lists^map/1-0-"/2
  (elixir) lib/enum.ex:1336: Enum."-map/2-lists^map/1-0-"/2
```

### Related issues

I don't think it's:

* #654 because I'm on 1.9.4
* #695 because I'm on 1.9.4
* #729 because I filtered out emoji

**Comment by garthk** [comment]
Quick repro without mucking up your environment:

```
git clone https://gist.github.com/garthk/42b156ae4ea47fbc8ec8d1fc8af69956
cd 42b156ae4ea47fbc8ec8d1fc8af69956
docker build -t credo-732 . && docker run --rm -ti credo-732
```

**Comment by TheFirstAvenger** [comment]
This seems to be an issue with Heredocs.replace_with_spaces. Using this test:

```
  test "issue #732" do
    foo =
    """
    defmodule InflictParserError do
      @moduledoc false

      def a_method do
        ~S(")
        ~S(])
        ~S([)
      end

      @doc \"\"\"
      \"\"\"
      def another_method, do: nil
    end
    """
    assert foo == foo
    |> to_source_file
    |> Heredocs.replace_with_spaces()
  end
```

you can see that it is replacing both `another_method` and the module `end` with spaces. The parsing error comes from that module `end` being wiped out.

```
  1) test issue #732 (Credo.Code.HeredocsTest)
     test/credo/code/heredocs_test.exs:701
     Assertion with == failed
     code:  assert foo == foo |> to_source_file |> Heredocs.replace_with_spaces()
     left:  "defmodule InflictParserError do\n  @moduledoc false\n\n  def a_method do\n    ~S(\")\n    ~S(])\n    ~S([)\n  end\n\n  @doc \"\"\"\n  \"\"\"\n  def another_method, do: nil\nend\n"
     right: "defmodule InflictParserError do\n  @moduledoc false\n\n  def a_method do\n    ~S(\")\n    ~S(])\n    ~S([)\n  end\n\n  @doc \"\"\"\n  \"\"\"\n                             \n   \n"
     stacktrace:
       test/credo/code/heredocs_test.exs:718: (test)
```

**Comment by rrrene** [comment]
Thx for reporting this! :+1: It should be be fixed in the latest RC of Credo (`v1.2.0-rc4`).

If it is not, please feel free to re-open this issue!
